### PR TITLE
Replace assert statements with raise exceptions

### PR DESCRIPTION
**Pull request by miyu386**
# What does this PR do?
Fixes an instance of https://github.com/huggingface/transformers/issues/12789.

Replaces 4 assert statements with ValueError exception in `src/transformers/data/metrics/squad_metrics.py`
Co-author: @mollerup23

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@sgugger

### CODE REVIEW

File: src/transformers/data/metrics/squad_metrics.py [hunk 1]
```diff
@@ -536,7 +536,8 @@ def compute_predictions_logits(
         if not nbest:
             nbest.append(_NbestPrediction(text="empty", start_logit=0.0, end_logit=0.0))
 
-        assert len(nbest) >= 1, "No valid predictions"
+        if not len(nbest) >= 1:
```
Comment (sgugger) [inline src/transformers/data/metrics/squad_metrics.py:539]:
```suggestion
        if len(nbest) < 1:
```

File: src/transformers/data/metrics/squad_metrics.py [hunk 2]
```diff
@@ -557,7 +558,8 @@ def compute_predictions_logits(
             output["end_logit"] = entry.end_logit
             nbest_json.append(output)
 
-        assert len(nbest_json) >= 1, "No valid predictions"
+        if not len(nbest_json) >= 1:
```
Comment (sgugger) [inline src/transformers/data/metrics/squad_metrics.py:561]:
```suggestion
        if len(nbest_json) < 1:
```

File: src/transformers/data/metrics/squad_metrics.py [hunk 3]
```diff
@@ -752,8 +754,10 @@ def compute_predictions_log_probs(
             output["end_log_prob"] = entry.end_log_prob
             nbest_json.append(output)
 
-        assert len(nbest_json) >= 1, "No valid predictions"
-        assert best_non_null_entry is not None, "No valid predictions"
+        if not len(nbest_json) >= 1:
```
Comment (sgugger) [inline src/transformers/data/metrics/squad_metrics.py:757]:
```suggestion
        if len(nbest_json) < 1:
```

**Review by sgugger — COMMENTED** [review]
Thanks for your PR, let's just make the negations more readable.

**Comment by miyu386** [comment]
@sgugger Appreciate the feedback! I've addressed and made the necessary changes.

**Review by sgugger — APPROVED** [review]
Thanks a lot!

**Pull request merged**